apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java

```diff
@@ -116,7 +116,7 @@ public void onClick(View view) {
 										Interaction interaction = interactions.getInteraction(interactionIdToLaunch);
 										if (interaction != null) {
 											String data = String.format("{\"title\":\"%s\",\"position\":%d,\"target\":\"%s\"}", buttonAction.getLabel(), position, interaction.getId());
-											EngagementModule.engageInternal(activity, interaction.getType().name(), TextModalInteraction.EVENT_NAME_INTERACTION, data);
+											EngagementModule.engageInternal(activity, interaction, TextModalInteraction.EVENT_NAME_INTERACTION, data);
```

Comment:
The com.apptentive#TextModal#interaction metric will not be sent if there are no valid Invocations that also result in a valid Interaction running.

Should the action metric be recorded "on button tap", regardless of the result of that button?